frontend/containers/UploadFileForm/UpdateFileForm.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { FC, useState } from "react";
+
+import { useRouter } from "next/dist/client/router";
+import { Input, Form, Button } from "nhsuk-react-components";
+import { useForm } from "react-hook-form/dist/index.ie11";
+import { Client } from "urql";
+
+import { Textarea } from "../../components/Textarea";
+import {
+  FileUploadUrlsDocument,
+  useCreateFileVersionMutation,
+} from "../../lib/generated/graphql";
+import { uploadBlob } from "../../lib/uploadBlob";
+import { useMaxLengthHelper } from "../../lib/useMaxLengthHelper";
+import {
+  StyledInput,
+  StyledTag,
+  StyledButton,
+  StyledFileInfoBox,
+  FormField,
+  StyledHeadingSection,
+  StyledFileName,
+} from "./styles";
+
+interface FormData {
+  files: FileList;
+  fileData: { title: string; description: string };
+}
+interface Props {
+  fileDescription: string;
+  fileId: string;
+  fileTitle: string;
+  folderId: string;
+  latestVersionId: string;
+  urqlClient: Client;
+  workspaceId: string;
+}
+
+const UpdateFileForm: FC<Props> = ({
+  fileDescription,
+  fileId,
+  fileTitle,
+  folderId,
+  latestVersionId,
+  urqlClient,
+  workspaceId,
+}) => {
+  const { register, handleSubmit, errors, setError } = useForm<FormData>();
+  const [, createFileVersion] = useCreateFileVersionMutation();
+
+  const router = useRouter();
+  const backToPreviousPage = () => router.back();
+
+  const titleMaxLength = useMaxLengthHelper("Title", 50);
+  const descriptionMaxLength = useMaxLengthHelper("Description", 250);
+
+  const [fileName, setFileName] = useState<string | undefined>();
+
+  const handleFile = (files: FileList | null) => {
+    setFileName(files?.[0]?.name);
+  };
+
+  const onSubmit = async (formData: FormData) => {
+    try {
+      const { error, data } = await urqlClient
+        .query(FileUploadUrlsDocument, { count: 1 })
+        .toPromise();
+      if (error) {
+        throw new Error(`Failed to get upload URL: ${error.toString()}`);
+      }
+
+      if (data) {
+        try {
+          await uploadBlob(data.fileUploadUrls[0], formData.files[0]);
+
+          const file = await createFileVersion({
+            newFileVersion: {
+              description: formData.fileData.description,
+              file: fileId,
+              fileName: formData.files[0].name,
+              fileType: formData.files[0].type,
+              folder: folderId,
+              latestVersion: latestVersionId,
+              temporaryBlobStoragePath: data.fileUploadUrls[0],
+              title: formData.fileData.title,
+            },
+          });
+
+          if (file.error) {
+            throw new Error(`Failed to save file: ${file.error?.message}`);
+          }
+
+          router.push(
+            `/workspaces/${workspaceId}/folders/${folderId}/files/${fileId}`
+          );
+        } catch (error) {
+          setError(`fileData.title`, {
```

Comment:
```suggestion
          setError("fileData.title", {
```
(backticks seem unnecessary. I've not read what setError is doing, so might be that this should be different in some other way)